picker-starter/src/picker.config.ts

```diff
@@ -1,34 +1,55 @@
-import { getItemFilters, queryItems } from '@/data'
-import { defineConfig } from '@/core'
+import { categoryMockAssets, items } from '@/data'
+import { defineConfig, matchCategories, matchItem } from '@/core'
 import { StoryblokIcon } from './components'
+import { getPage } from './utils'
 
-export default defineConfig((options) => {
-  return {
-    title: 'Picker Starter',
-    icon: StoryblokIcon,
-    validateOptions: () => {
-      const { limit } = options
+export default defineConfig((options) => ({
+  title: 'Picker Starter',
+  icon: StoryblokIcon,
+  validateOptions: () => {
+    const { limit } = options
 
-      const isLimitOptionValid = limit === undefined || Number(limit) > 0
-
-      if (!isLimitOptionValid) {
-        return {
-          isValid: false,
-          error: `The 'limit' option must be an integer greater than 0`,
-        }
-      }
+    const isLimitOptionValid = limit === undefined || Number(limit) > 0
 
+    if (!isLimitOptionValid) {
       return {
-        isValid: true,
+        isValid: false,
+        error: `The 'limit' option must be an integer greater than 0`,
       }
-    },
-    tabs: [
-      {
-        name: 'items',
-        label: 'Items',
-        query: queryItems,
-        getFilters: getItemFilters,
+    }
+
+    return {
+      isValid: true,
+    }
+  },
+  tabs: [
+    {
+      name: 'items',
+      label: 'Items',
+      query: async ({ searchTerm, page, perPage, filterSelection }) => {
+        const filteredItems = items
+          .filter(matchCategories(filterSelection['categoryMulti'] as string[]))
+          .filter(matchItem(searchTerm))
```

Comment:
`matchItem` sounds a bit unintuitive to me. It can be something like `matchSearchTerm`...?

Or we could rename them altogether like:

```js
        const filteredItems = items
          .filter(filterByCategories(filterSelection['categoryMulti'] as string[]))
          .filter(filterBySearchTerm(searchTerm))
```

What do you think? 🤔

Reply:
> matchItem sounds a bit unintuitive to me. It can be something like matchSearchTerm...?

Yeah, you're totally right, `matchSearchTerm` is much better.

About the `filterByCategories` and `filterBySearchTerm` renaming, I'm not too sure.
These names suggest to me this function is going to perform the filtering action while it doesn't.

I don't know... somehow using `match...` was more meaningful to me.

What do you think, @eunjae-lee?

Reply:
okay, then it sounds good. let's merge!